assignments/lambda-classes.js

```diff
@@ -1 +1,110 @@
 // CODE here for your Lambda Classes
+
+class Person {
+  constructor(element) {
+    this.name = element.name;
+    this.age = element.age;
+    this.location = element.location;
+    this.gender = element.gender;
+  }
+  speak() {
+    return `Hello my name is ${this.name}, I am from ${this.location}`;
+  }
+}
+
+class Instructor extends Person {
+  constructor(element) {
+    super(element);
+    this.specialty = element.specialty;
+    this.favLanguage = element.favLanguage;
+    this.catchPhrase = element.catchPhrase;
+  }
+
+  demo(subject) {
+    console.log(`Today we are learning about ${subject}`);
+  }
+
+  grade(student, subject) {
+    console.log(`${student.name} receives a perfect score on ${subject}`);
+  }
+}
+
+class Student extends Person {
+  constructor(element) {
+    super(element);
+    this.previousBackground = element.previousBackground;
+    this.className = element.className;
+    this.favSubjects = element.favSubjects;
+  }
+
+  listsSubjects() {
+    this.favSubjects.forEach(element => {
```

Comment:
```
Cannot read property 'forEach` of undefined
```

I refractored it to 
```
listsSubjects(subjects) {
    subjects.forEach(element => {
      console.log(element);
    });
  }
```

make sure you are passing an array and because you are, you must pass it inside the parameter